.travis.yml

```diff
@@ -5,18 +5,19 @@
 # we will use conda to give us a much faster setup time.
 
 
-language: python
-python:
-  - 2.7
-  - 3.6
+language: minimal
 sudo: false
 
 env:
   matrix:
-    - TEST_TARGET=default TEST_MINIMAL=true
-    - TEST_TARGET=default
-    - TEST_TARGET=example
-    - TEST_TARGET=doctest
+    - PYTHON_VERSION=2.7 TEST_TARGET=default TEST_MINIMAL=true
+    - PYTHON_VERSION=3.7 TEST_TARGET=default TEST_MINIMAL=true
+    - PYTHON_VERSION=2.7 TEST_TARGET=default
+    - PYTHON_VERSION=3.7 TEST_TARGET=default
+    - PYTHON_VERSION=2.7 TEST_TARGET=example
+    - PYTHON_VERSION=3.7 TEST_TARGET=example
+    - PYTHON_VERSION=3.7 TEST_TARGET=doctest
+    # doctests only with Python3, see https://github.com/SciTools/iris/pull/3134
```

Comment:
The downside, for me, is that really we should be covering py27, py36 and py37... At this point, we've lost the convenience of the matrix...

Reply:
The alternative that I know works with Travis-CI is not better than this. It would require something like:

```yaml
python:
- 2.7
- 3.6

matrix:
  include:
    - python: 3.7
      dist: xenial
      sudo: true
```

Reply:
Maybe some reordering into blocks by python version can make this more appealing:
```yaml
env:
  matrix:
    - PYTHON_VERSION=2.7 TEST_TARGET=default TEST_MINIMAL=true
    - PYTHON_VERSION=2.7 TEST_TARGET=default
    - PYTHON_VERSION=2.7 TEST_TARGET=example
    # doctests only with Python3, see https://github.com/SciTools/iris/pull/3134
    #- PYTHON_VERSION=2.7 TEST_TARGET=doctest

    - PYTHON_VERSION=3.6 TEST_TARGET=default TEST_MINIMAL=true
    - PYTHON_VERSION=3.6 TEST_TARGET=default
    - PYTHON_VERSION=3.6 TEST_TARGET=example
    - PYTHON_VERSION=3.6 TEST_TARGET=doctest

    - PYTHON_VERSION=3.7 TEST_TARGET=default TEST_MINIMAL=true
    - PYTHON_VERSION=3.7 TEST_TARGET=default
    - PYTHON_VERSION=3.7 TEST_TARGET=example
    - PYTHON_VERSION=3.7 TEST_TARGET=doctest
```